### ISSUE TITLE
cant get the basics working

### ISSUE THREAD

**Issue by rifaterdemsahin** [MScholtes/PSVirtualDesktop#29]
<img width="552" alt="image" src="https://github.com/MScholtes/PSVirtualDesktop/assets/5270036/a164d3f9-3421-4e9a-bad4-99493dfee8e5">

**Comment by rifaterdemsahin** [comment]
<img width="871" alt="image" src="https://github.com/MScholtes/PSVirtualDesktop/assets/5270036/a391aed0-b141-4e78-bbbf-14ef4f239e53">

**Comment by MScholtes** [comment]
Hello @rifaterdemsahin,

I guess you are on a _Windows Insider_ version. There is limited support.

Can you tell which Insider version and post the build number (you can display it with _WinVer_ or in _Powershell_ with _$PSVersionTable_)?

Greetings

Markus

**Comment by MScholtes** [comment]
OK, I have made two attempts to find out what changed:

The first attempt (attached VirtualDesktopV3.ps1) assumes that the API is the same from new Windows 11 build 22621 upwards.

The second attempt (attached VirtualDesktopV4.ps1) assumes that the API from new build 22621 onwards is the same as the API from Windows Insider Beta before 8 March 2023 and that another API applies from Insider Dev onwards.

Please remove the additional extension txt from the files and try the following in the directory of the scripts:

```powershell
. .\VirtualDesktopV3.ps1
Get-DesktopList
```

exit Powershell now, start a new Powershell session and then

```powershell
. .\VirtualDesktopV4.ps1
Get-DesktopList
````

Can you post what's happening?

Greetings

Markus
[VirtualDesktopV4.ps1.txt](https://github.com/MScholtes/PSVirtualDesktop/files/12426011/VirtualDesktopV4.ps1.txt)
[VirtualDesktopV3.ps1.txt](https://github.com/MScholtes/PSVirtualDesktop/files/12426012/VirtualDesktopV3.ps1.txt)

**Comment by MScholtes** [comment]
I have published the new version 1.5.4 of VirtualDesktop in the Powershell Gallery. However, I am not sure whether all versions and commandlets work.

I ask for feedback.

Greetings

Markus

**Comment by MScholtes** [comment]
Hello,

I released a new bug fix version 1.5.5 today - tested only against Windows 11 22H2.2215. 
According to Microsoft's will, this may become obsolete again with the September update.

It corrects issues with creating, removing and naming desktops on Windows 11 22H2.2215 (reason for the error: _SwitchDesktopAndMoveForegroundView_ is missing in 2215).

I do not know, if this works on Insider versions build 22631 and up. It would be helpful if you try
```powershell
New-Desktop
```
if you own an Insider version and post the result and the build number here.


Greetings

Markus

**Comment by MScholtes** [comment]
No further activities required
